vertical-pod-autoscaler/pkg/admission-controller/config.go

```diff
@@ -141,9 +141,15 @@ func selfRegistration(clientset kubernetes.Interface, caCert []byte, webHookDela
 			},
 		}
 	}
+	webhookLabelsMap, err := convertLabelsToMap(webHookLabels)
+	if err != nil {
+		klog.Warningf("Unable to parse webhook labels. Creating webhook without labels: %v\n", err)
```

Comment:
We're switching to structured logging currently (see https://github.com/kubernetes/autoscaler/pull/7226 for the admission-controller PR), therefore I'd like this to follow the same pattern. As warnings don't exist in structured logging, this probably should be logged with `klog.ErrorS`?

I'm also wondering if we should continue with the selfRegistration here or not? Currently this error just goes into the logs and everything still get registered – is this intentional?

Reply:
I change it to `klog.ErrorS`. I think use labels is an option but not necessary, so when cx use invalid labels format, selfRegistration will continue and just print a error log.